### PR TITLE
refactor(hooks): port useSpring + useCycle to Svelte 5; ExampleV2 docs migration

### LINKED ISSUE #358: Port `useCycle` to Svelte 5 runes
The current implementation in [`src/lib/utils/cycle.ts`](https://github.com/humanspeak/svelte-motion/blob/main/src/lib/utils/cycle.ts) uses a `writable` store from `svelte/store` which is the Svelte 4 way of doing things.

Solution: 
The file should be renamed to `cycle.svelte.ts` to signal it uses runes.

New Code: 
```ts
import { wrap } from 'motion'

export type Cycle = (next?: number) => void
export type CycleState<T> = { readonly current: T; cycle: Cycle }

/**
 * Cycles through a series of values. Mirrors Framer Motion's `useCycle`.
 *
 * Function-based Svelte 5 port: returns a plain object with a `$state`-backed
 * `current` property and a `cycle` function. Templates read `state.current`
 * directly — no `$` store prefix needed.
 *
 * The tuple API from the Svelte 4 version is intentionally dropped: runes
 * must be defined at the top level of a `.svelte.ts` file, not destructured
 * from a returned array, or reactivity is lost.
 *
 * @param items - Items to cycle through. Must include at least one item.
 * @returns A `{ current, cycle }` object.
 * @see https://motion.dev/docs/react-use-cycle
 *
 * @example
 * ```svelte
 * <script>
 *   import { useCycle } from '@humanspeak/svelte-motion'
 *
 *   const x = useCycle(0, 50, 100)
 * </script>
 *
 * <motion.div animate={{ x: x.current }} onclick={() => x.cycle()} />
 * ```
 */
export const useCycle = <T>(...items: T[]): CycleState<T> => {
    if (items.length === 0) {
        throw new Error('useCycle requires at least one item')
    }

    let index = 0
    let current: T = $state(items[0])

    const cycle: Cycle = (next?: number) => {
        const target =
            typeof next === 'number' ? next : wrap(0, items.length, index + 1)
        if (target === index) return
        index = target
        current = items[target]
    }

    return {
        get current() {
            return current
        },
        cycle,
    }
}
```
Change in output: 
Return `{ current, cycle }` instead of the `[store, cycle]` tuple (destructuring breaks reactivity with runes)

### LINKED ISSUE #361: [W1] useSpring → MotionValue + .current
**Wave 1 anchor — Tier 2 hybrid.** Landed in #360.

Replaces the hand-rolled Svelte-store + Hooke's-law spring with a real motion-dom `MotionValue` driven by `attachFollow`, augmented with a `$state`-backed `.current` getter and a `.subscribe()` compat shim.

## What landed

- Returns `SpringMotionValue<T>` = motion-dom `MotionValue<T>` + `.current` (Svelte 5 reactive) + `.subscribe(run)` (Svelte readable contract)
- Full `SpringOptions` surface incl. `skipInitialAnimation`, `duration`, `visualDuration`, `bounce`, `velocity`
- Velocity carryover on retarget, unit-string preservation, `animationStart`/`animationComplete` events — free via motion-dom
- 13 unit tests covering identity, jump, subscribe shim, source forms, lifecycle
- Demo route `/tests/use-spring`

## Files touched

- `src/lib/utils/spring.svelte.ts` (new)
- `src/lib/utils/spring.svelte.spec.ts` (new)
- `src/lib/utils/spring.ts` (deleted)
- `src/lib/utils/spring.spec.ts` (deleted)
- `src/lib/index.ts` (re-export)
- `src/routes/+page.svelte` (link)
- `src/routes/tests/use-spring/+page.svelte` (demo)

Resolved via #360.

### LINKED ISSUE #362: [W1] useCycle → { current, cycle }
**Wave 1 anchor — Tier 1.** Landed in #360. Closes #358.

Replaces `[Readable<T>, Cycle]` tuple with a `{ current, cycle }` object — destructuring a `$state`-backed tuple snapshots the value and loses reactivity under runes, which is the root cause #358 flagged.

## What landed

- Two call forms: `useCycle(...items)` (matches React) and `useCycle(() => items)` (new — reactive items overload)
- Full-range clamp on read (negative → first, overflow → last)
- Integer-only `cycle(next)` guard (throws on `NaN` / `Infinity` / fractional)
- Reactive-getter empty-mid-cycle → `.current` throws
- 23 unit tests covering all forms + edge cases

## Files touched

- `src/lib/utils/cycle.svelte.ts` (new)
- `src/lib/utils/cycle.svelte.spec.ts` (new)
- `src/lib/utils/cycle.ts` (deleted)
- `src/lib/utils/cycle.spec.ts` (deleted)
- `src/lib/index.ts` (re-export)
- `src/routes/tests/cycle/+page.svelte` (consumer)
- `docs/src/lib/examples/use-cycle/demos/Default.svelte` (consumer)

Resolved via #360.

### PR DESCRIPTION
**Pull request by jaysin586**
## Summary

Two threads landed on this branch:

1. **ExampleV2 docs migration** (the original branch scope) — every docs example route rebuilt on `@humanspeak/docs-kit`'s new `ExampleV2` shell, with the demo files restructured to fit docs-kit 2026.5.31+ ergonomics.
2. **Svelte 5 hook migration (Tier 1)** — port `useSpring` and `useCycle` off Svelte stores onto runes-native shapes. Triggered by #358 and described in the comment thread there; this PR delivers the first wave of that plan.

The two are entangled because the new `useCycle` API drives the use-cycle example page that was being migrated in parallel.

### ✨ Svelte 5 hook migration

- **`useSpring`** — full rewrite. Now returns a real motion-dom `MotionValue` (so it composes with `animate()` and the rest of motion-dom natively) augmented with a Svelte-5-reactive `.current` getter and a `.subscribe()` shim for legacy `$store` template syntax and `useTransform([store])` consumers. Spring physics delegated to motion-dom's `attachFollow`/`JSAnimation`, giving us 1:1 React parity for option semantics (full `SpringOptions` + `skipInitialAnimation`), velocity carryover on retarget, unit-string preservation, and `animationStart`/`animationComplete` events for free.
- **`useCycle`** — replaced the `[Readable<T>, Cycle]` tuple with a `{ current, cycle }` object that works under runes (destructuring a `$state`-backed tuple snapshots the value and loses reactivity — that's the issue #358 root cause). Two call forms: `useCycle(...items)` matches React framer-motion 1:1; new `useCycle(() => items)` overload supports reactive item lists (Svelte's once-per-mount hook model otherwise can't pick up prop/derived changes the way React's per-render hooks can).
- Demo pages at `/tests/use-spring` and `/tests/cycle` exercise every API surface; both linked from the root `+page.svelte` index.

### 🔧 ExampleV2 docs migration

- 25 docs example pages migrated from V1 `Example` to docs-kit `ExampleV2`:
  animate-presence, animated-button, html-content, hover-and-tap, motion-path, notifications-stack, rotate, scroll-progress, shared-layout-animation, style-string, use-animate, use-animation-frame, use-cycle, use-in-view, use-presence, use-reduced-motion, use-time, use-time-synced, variants-basic, variants-propagation, variants-while-hover, while-focus, while-in-view, and the kickoff batch.
- Demo files refactored to the docs-kit 2026.5.31 ergonomics shape.
- Old Batch 1 monolithic example files deleted.

### 📚 Documentation

- Rewrote `/docs/use-spring` — correct defaults (motion-dom's `100/10/1`), `.current` as the primary read pattern, full `SpringOptions` surface (`duration`, `visualDuration`, `bounce`, `velocity`, `skipInitialAnimation`), motion-dom delegation explained in "How it works".
- Rewrote `/docs/use-cycle` — `{ current, cycle }` shape, reactive-items section, divergence note vs React's tuple.
- Forward-looking `/docs/motion-values` overview with a Tier 1/Tier 2 migration matrix showing which hooks now return motion values vs which still expose Svelte stores.
- `llms-full.txt` `useSpring`/`useCycle` sections updated to match.
- Touched-up snippets in `/docs/use-scroll` and `/docs/use-motion-value-event` to use `.current` on springs.

### 📦 Dependencies

- `@humanspeak/docs-kit` `2026.5.31` → `2026.5.32` (gains markdown-rendering in `ExampleV2.description` so backticks render as styled code chips).
- `@humanspeak/svelte-markdown` `^1.5.2` added as a direct dep on `docs/` (docs-kit declares it as an optional peer).
- `motion` + `motion-dom` `^12.39.0` → `^12.40.0` (semver-minor, test suites green).
- Lockfile refreshed; trunk-managed `trunk-io/plugins` `v1.9.0` → `v1.10.0`, `eslint` `10.3.0` → `10.4.0`.

### 🔄 Tooling

- `chore(trunk)`: ignore `docs/static/r/**` — the shadcn-style registry blobs are generated/curated externally with a non-default indent, and trunk was rewriting them on every fmt run.
- `chore(gitignore)`: ignore `.claude/scheduled_tasks.lock`.

## Breaking changes

Both Tier 1 hooks change shape. Migration guides are in their respective `/docs/` pages; quick version:

- `useCycle`: replace `const [x, cycleX] = useCycle(...)` + `$x` with `const x = useCycle(...)` + `x.current` and `x.cycle()`.
- `useSpring`: existing `$spring` syntax still works via the `.subscribe()` shim, so no forced changes; the recommended pattern is `spring.current`. Default physics shift from svelte-motion's prior `170/26` to motion-dom's `100/10/1` (matches React framer-motion) — bouncier feel for callers using `useSpring(value)` with no options.

## Test plan

- [x] `pnpm vitest run` — useSpring (13) + useCycle (17) suites green; full suite green on motion 12.40.0
- [x] `pnpm check` — 0 errors
- [x] `pnpm package` + `publint` clean
- [x] Demo routes (`/tests/use-spring`, `/tests/cycle`) visually verified in browser; cycle button advances, spring overshoots and settles
- [x] Docs site (`/docs/use-spring`, `/docs/use-cycle`, `/examples/use-cycle`) renders correctly with description-markdown code chips
- [ ] CodeRabbit review on the diff
- [ ] Manual smoke test of the broader ExampleV2 migration pages

Closes #358

🤖 Generated with [Claude Code](https://claude.com/claude-code)

Closes #361

Closes #362

### CODE REVIEW

File: docs/src/lib/examples/hover-and-tap/demos/Default.svelte [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+</script>
+
+<!-- dk-strip: docs-kit positioning shell — stripped from the published code. -->
+<div class="dk-demo-shell">
+    <motion.button
+        class="text-text-primary-muted h-[100px] w-[100px] rounded-md border border-border bg-green-500 px-4 py-2"
+        whileHover={{ scale: 1.2 }}
+        whileTap={{ scale: 0.8 }}
+    ></motion.button>
+    <span class="hint">hover · tap</span>
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply docs/src/lib/examples/hover-and-tap/demos/Default.svelte:18]: Fixed — added `type="button"` + `aria-label="Hover or tap to scale"` on the `motion.button`.

File: docs/src/lib/examples/notifications-stack/demos/Default.svelte [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+                    type: 'spring',
+                    stiffness: 600,
+                    damping: 50,
+                    delay: i * 0.04
+                }}
+                style={getNotificationStyle(i)}
+                onclick={() => (isOpen = !isOpen)}
+            >
+                Note {i + 1}
+            </motion.div>
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply docs/src/lib/examples/notifications-stack/demos/Default.svelte:152]: Fixed — kept `motion.div` and added `role="button"`, `tabindex="0"`, `aria-pressed={isOpen}`, `aria-label="Toggle notification stack"`, plus an Enter/Space keydown handler that mirrors the onclick. I went with the role/keydown approach rather than swapping to `motion.button` so the existing `variants` + inline-style pipeline does not have to fight default button styling on stacked cards.

File: docs/src/lib/examples/rotate/demos/Default.svelte [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+</script>
+
+<!-- dk-strip: docs-kit positioning shell — stripped from the published code. -->
+<div class="dk-demo-shell">
+    {#key replayId}
+        <motion.ul
+            class="size-24 rounded-md bg-pink-400"
+            animate={{ rotate: 360 }}
+            transition={{ duration: 1 }}
+        />
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply docs/src/lib/examples/rotate/demos/Default.svelte]: Fixed — swapped `motion.ul` for `motion.div` on the decorative square.

File: docs/src/lib/examples/shared-layout-animation/demos/Default.svelte [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+                            backgroundColor:
+                                i === selectedTab
+                                    ? 'color-mix(in srgb, var(--color-card-foreground) 10%, transparent)'
+                                    : 'transparent'
+                        }}
+                        class="tab"
+                        role="tab"
+                        aria-selected={i === selectedTab}
+                    >
+                        <button class="tab-button" onclick={() => (selectedTab = i)}>
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply docs/src/lib/examples/shared-layout-animation/demos/Default.svelte]: Fixed — moved `role="tab"` + `aria-selected` onto the focusable `<button>`, added `role="tablist"` on the wrapping `<ul>`, and marked the `motion.li` with `role="presentation"` so the structural element does not announce as a duplicate tab.

File: docs/src/routes/docs/motion-values/+page.svx [hunk 5]
```diff
@@ -126,14 +173,25 @@ Subscribe to value changes with `useMotionValueEvent` or directly with `store.su
...
 
+Springs additionally expose motion-dom event types via `.on(...)`:
+
+```svelte
+<script>
+  const x = useSpring(0)
+
+  x.on('animationStart', () => console.log('spring started'))
+  x.on('animationComplete', () => console.log('spring settled'))
+</script>
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply docs/src/routes/docs/motion-values/+page.svx:186]: Fixed — added `import { useSpring } from '@humanspeak/svelte-motion'` to the animation-events snippet so it copy/pastes cleanly.

File: docs/src/routes/examples/use-cycle/+page.svelte [hunk 6]
```diff
@@ -11,14 +20,92 @@
...
+                for tab-bar-style controls where each button maps to a specific state — no need for
+                a separate selected-index <code>$state</code> cell.
+            </span>
+        </li>
+        <li>
+            <ListChecks />
+            <span>
+                The motion target reads
+                <code>animate={'{variant.current}'}</code> (a string label), so motion resolves it
+                via <code>variants</code>. Pure declarative: state lives in the cycle object,
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply docs/src/routes/examples/use-cycle/+page.svelte:76]: Fixed — standardized on `variant.current` / `variant.cycle()` throughout the example prose. `variant` reads as more semantic than `state` for a hook that cycles through animation variants, so I pulled the description text in to match the notes instead of the other way around.

File: src/lib/utils/cycle.svelte.spec.ts [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+
+    it('jumps to a specific index when called with a number', () => {
+        inRoot(() => {
+            const x = useCycle(10, 20, 30, 40)
+            x.cycle(2)
+            expect(x.current).toBe(30)
+            x.cycle(0)
+            expect(x.current).toBe(10)
+        })
+    })
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply src/lib/utils/cycle.svelte.spec.ts:66]: Added 4 edge tests under a new `out-of-range cycle(i)` describe block: positive overflow clamps to the last item, negative indexes return `undefined` (the clamp predicate is `index >= items.length`, so this is the documented gap), `cycle()` after an OOR jump wraps via `wrap(0, len, stored + 1)`, and jumping back into range resolves cleanly. The negative-index test asserts the actual behavior so it fails loudly if the clamp predicate ever changes.

File: src/lib/utils/cycle.svelte.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+ *   list changes without recreating it.
+ *
+ * In both forms:
+ *
+ * - `state.current` is reactive — read it in templates / `$derived` / `$effect`
+ *   and it tracks both index changes and (in the getter form) item changes.
+ * - `state.cycle()` advances to the next item (wrapping at the end).
+ * - `state.cycle(i)` jumps to index `i`. Out-of-range indexes pass through
+ *   to `items[i]` (which may be `undefined`) — same as React framer-motion.
+ * - Calls that resolve to the current index are no-ops, matching React
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply src/lib/utils/cycle.svelte.ts]: Aligned in favor of the clamp — kept it because removing it would make the reactive-getter form (`useCycle(() => items)`) return `undefined` whenever `items` shrinks below the stored index, which defeats most of the reason for that overload existing. Updated the JSDoc on `cycle.svelte.ts`, the `/docs/use-cycle` page, and the `llms-full.txt` section to document the clamp explicitly and call it out as a deliberate divergence from React framer-motion (which returns `items[i]`, possibly undefined). Also documented that subsequent `cycle()` advances are relative to the *stored* (unclamped) index, so `cycle(99); cycle()` on a 3-item cycle wraps via `wrap(0, 3, 100) = 1` rather than continuing from the clamped read position. The 4 new edge tests in the spec lock this contract.

File: src/lib/utils/spring.svelte.ts [hunk 9]
```diff
@@ -0,0 +1,237 @@
...
+    options?: UseSpringOptions
+): SpringMotionValue<string>
+export function useSpring(
+    source: Readable<number | string>,
+    options?: UseSpringOptions
+): SpringMotionValue<number> | SpringMotionValue<string>
+export function useSpring(
+    source: number | string | MotionValue<number> | MotionValue<string> | Readable<number | string>,
+    options: UseSpringOptions = {}
+): SpringMotionValue<number> | SpringMotionValue<string> {
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply src/lib/utils/spring.svelte.ts]: Fixed — replaced the `Readable<number | string>` overload with a generic `useSpring<T extends number | string>(source: Readable<T>, options?): SpringMotionValue<T>`. Callers passing `Readable<number>` now get `SpringMotionValue<number>` back instead of the widened union.

File: src/lib/utils/cycle.svelte.ts [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+ * In both forms:
+ *
+ * - `state.current` is reactive — read it in templates / `$derived` / `$effect`
+ *   and it tracks both index changes and (in the getter form) item changes.
+ * - `state.cycle()` advances to the next item (wrapping at the end).
+ * - `state.cycle(i)` jumps to index `i`. The index is stored as-given; `.current`
+ *   then clamps on read so out-of-range indexes resolve to the last valid item
+ *   instead of `undefined`. This is a defensive divergence from React
+ *   framer-motion (which returns `items[i]`, possibly undefined) — needed so
+ *   the reactive-getter form stays safe when items shrink underneath the cycle.
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply src/lib/utils/cycle.svelte.ts]: Fixed properly this time — the previous pass left a partial clamp that only caught overflow. `.current` now clamps the full range (negative → `items[0]`, overflow → `items[length - 1]`) and throws on empty so the `T` return type holds in every reachable read path. The reactive-getter empty-mid-cycle case explicitly throws rather than leaking `undefined`.

File: src/lib/utils/cycle.svelte.spec.ts [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+            inRoot(() => {
+                const x = useCycle('a', 'b', 'c')
+                x.cycle(-5)
+                // Negative index is < items.length, so the clamp guard doesn't
+                // fire — `items[-5]` is undefined on a real array, but the
+                // clamp branch only catches index >= items.length. Document
+                // the actual behavior so the test fails loudly if the impl
+                // ever changes the clamp predicate.
+                expect(x.current).toBeUndefined()
+            })
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply src/lib/utils/cycle.svelte.spec.ts]: Fixed — renamed to `.current clamps negative indexes to the first item` and the assertion is now `expect(x.current).toBe('a')`. Also added a companion test that the reactive-getter empty-mid-cycle case throws, so the full clamp contract is locked in the spec.

File: docs/src/routes/docs/use-cycle/+page.svx [hunk 12]
```diff
@@ -63,61 +60,100 @@ the `animate` prop:
...
+
+- **`current`** `T` (getter) — the current item, starting at `items[0]`. Tracks reactively via `$state`. Clamps to the last valid index if items shrink underneath.
+- **`cycle(next?: number)`** — advance to the next item, or jump to the value at index `next` when supplied. No-ops if the resolved index is unchanged.
+
+### Notes
 
-- **value** `Readable<T>` &mdash; the current item, starting at `items[0]`.
-- **cycle** `(next?: number) => void` &mdash; advance to the next item, or jump
-  to the value at index `next` when supplied.
+- Out-of-range indexes are stored as-given by `cycle(i)` but `.current` always clamps on read — `x.cycle(99)` on a 3-item cycle yields `items[2]`, not `undefined`. This is a defensive divergence from React framer-motion (which returns `items[99]`) so the reactive-getter form stays safe when items shrink underneath. Subsequent `cycle()` advances are relative to the *stored* index, so `cycle(99); cycle()` on a 3-item cycle wraps via `wrap(0, 3, 100) = 1`.
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply docs/src/routes/docs/use-cycle/+page.svx]: Fixed — rewrote the divergence callout as two explicit numbered bullets (return shape and out-of-range clamp) so the "otherwise 1:1" sentence no longer contradicts the out-of-range section further down. Also updated the same wording in `llms-full.txt`.

**Comment by jaysin586** [comment]
@coderabbitai responding to your outside-diff review on `src/lib/utils/cycle.svelte.ts`:

- **`cycle(NaN)` / non-integer guard** — Fixed in `f108444`. `cycle()` now throws `useCycle index must be a finite integer` when `next` is provided but isn't an integer. A single `!Number.isInteger(next)` check covers `NaN`, `Infinity`, and fractional values in one branch (since `Number.isInteger` returns false for all three). Added a spec test that asserts the throw for all three inputs and that the original index is unchanged afterward. Good catch — the read-time clamp predicates (`< 0` and `>= length`) all return false for `NaN`, so it would have slipped straight through and made `.current` resolve to `undefined`.

- **JSDoc `@param` / `@returns` tags** — Added in `f108444`. Both overloads documented, plus the return-type contract notes the clamp/throw behaviors so consumers can find them without reading the impl.

- **Arrow-syntax conversion** — Intentionally skipped. `useCycle` uses TypeScript function-declaration overloads for the same reason `useSpring` does in this PR — it's the cleanest way to express multi-signature hooks. Doing the same with arrow syntax requires a separate overloaded type alias plus a cast on the implementation, which adds boilerplate without changing behavior. `useSpring` wasn't flagged, so I'm treating the convention as having an implicit exception for overload-bearing exports rather than churning this file alone for stylistic uniformity. Happy to revisit if you feel strongly.

File: src/lib/utils/cycle.svelte.ts [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+ * @param itemsGetter Function returning the current items list; re-invoked
+ *   on every `.current` read so reactive sources propagate. Use this form
+ *   when items can change between mount and unmount. (Reactive overload.)
+ * @param items One or more values to cycle through. Captured once at call
+ *   time and fixed for the cycle's lifetime. (Varargs overload.)
+ * @returns A `CycleState<T>` with a reactive `.current` getter and a
+ *   `cycle(next?: number)` advance/jump function. `.current` always
+ *   honors its `T` type by clamping out-of-range indexes; `.cycle()`
+ *   throws on non-integer (`NaN`, `1.5`, `Infinity`) indexes and on a
+ *   reactive getter that empties mid-cycle.
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply src/lib/utils/cycle.svelte.ts]: Fixed in `0063ead` — re-attributed the throws in the `@returns` block: `.current` throws if a reactive getter empties the items list mid-cycle; `.cycle()` throws on non-integer indexes and returns early as a no-op on empty items. The previous wording conflated the two throw sites.

File: docs/static/llms-full.txt [hunk 14]
```diff
@@ -584,18 +591,53 @@ Track whether an element is in the viewport (IntersectionObserver under the hood
...
+reactive-getter list that shrinks underneath the cycle is treated the
+same way. This is a defensive divergence from React framer-motion
+(which returns `items[i]`, possibly undefined) so `.current` always
+honors its `T` type. Subsequent `cycle()` advances are relative to the
+stored (unclamped) index, so `cycle(99); cycle()` wraps via
+`wrap(0, len, 100)`. If the reactive getter empties mid-cycle,
+`.current` throws rather than silently returning `undefined`.
+`cycle(next)` itself throws on non-integer indexes (`NaN`, `1.5`,
+`Infinity`) since those slip past the read-time clamp and would
+otherwise leak `undefined` through a `T`-typed read.
```
[the remark this answers is not in the document]
Comment (jaysin586) [reply docs/static/llms-full.txt:630]: Fixed in `0063ead` — promoted the divergence callout in both `/docs/use-cycle` and `llms-full.txt` from "two deliberate divergences" to three, with the integer-only `cycle(next)` constraint as the new bullet. The "otherwise 1:1" closing line now follows a complete list, so the parity claim is honest again.
